File: src/phorge/api/endpoints/sites.py
"""Site API endpoints."""

from __future__ import annotations

from phorge.api.client import ForgeClient
from phorge.api.models import Site


class SitesAPI:
    def __init__(self, client: ForgeClient) -> None:
        self._client = client

    async def list(self, server_id: int) -> list[Site]:
        data = await self._client.get(f"/servers/{server_id}/sites")
        return [Site(**s) for s in data.get("sites", [])]

    async def get(self, server_id: int, site_id: int) -> Site:
        data = await self._client.get(f"/servers/{server_id}/sites/{site_id}")
        return Site(**data["site"])

    async def update_aliases(self, server_id: int, site_id: int, aliases: list[str]) -> Site:
        data = await self._client.put(
            f"/servers/{server_id}/sites/{site_id}/aliases",
            json={"aliases": aliases},
        )
        return Site(**data["site"])

    async def change_php_version(self, server_id: int, site_id: int, version: str) -> None:
        await self._client.put(
            f"/servers/{server_id}/sites/{site_id}/php",
            json={"version": version},
        )

File: internal/forge/sites.go
package forge

import (
	"context"
	"net/http"
)

// List returns all sites on a server.
func (s *SitesService) List(ctx context.Context, serverID int64) ([]Site, error) {
	return listResources[Site](s.client, ctx, s.client.orgPath("/servers/%d/sites", serverID))
}

// Get returns a single site by ID.
func (s *SitesService) Get(ctx context.Context, serverID, siteID int64) (*Site, error) {
	return getResource[Site](s.client, ctx, s.client.orgPath("/sites/%d", siteID))
}

// UpdateAliases updates the domain aliases for a site.
func (s *SitesService) UpdateAliases(ctx context.Context, serverID, siteID int64, aliases []string) (*Site, error) {
	path := s.client.orgPath("/servers/%d/sites/%d", serverID, siteID)
	body := map[string]any{"aliases": aliases}
	return updateResource[Site](s.client, ctx, path, body)
}

// UpdatePHP changes the PHP version for a site.
func (s *SitesService) UpdatePHP(ctx context.Context, serverID, siteID int64, version string) error {
	path := s.client.orgPath("/servers/%d/sites/%d", serverID, siteID)
	body := map[string]string{"php_version": version}
	return s.client.do(ctx, http.MethodPut, path, body, nil)
}
